ddtrace/tracer/rules_sampler.go

```diff
@@ -589,78 +590,67 @@ func samplingRulesFromEnv() (trace, span []SamplingRule, err error) {
 	return trace, span, err
 }
 
-// unmarshalSamplingRules unmarshals JSON from b and returns the sampling rules found, attributing
-// the type t to them. If any errors are occurred, they are returned.
-func unmarshalSamplingRules(b []byte, spanType SamplingRuleType) ([]SamplingRule, error) {
+func (sr *SamplingRule) UnmarshalJSON(b []byte) error {
```

Comment:
I would suggest you consolidate behaviour in one place, otherwise with time the disparity between the implementations will grow (for example, right now the RC change rules and Environment setup rules would behave differently)